# Instagram Likes Report

By: Thomas Laryea

September 15th, 2023

My hypothesis for this data is my likes will reflect a general bias towards sports related content. I also believe there will be a fair mixture of "meme" related content. This is due to the fact of the longevity of my instagram account where when I was younger I followed pages geared more towards memes and as I grew older I feel my likes and interests geared more towards sports and the content associated with it. I expect to see within the data sports related content to make up more than 70% of my likes in the top 50 of accounts I've liked.

The data used within this report was recorded by Instagram, and by proxy, Meta, it is from my personal Instagram account. This data was most likely collected to serve as information for advertisers. My likes and interests can be sold to brands looking to sell their products, and Instagram happens to have a lot of this information. This data is reliable because it is about me and my interests; there is no outside interference. However, it may be potentially unreliable in the sense that it is potentially dated. I have used the same Instagram account since 6th grade, and my interests have changed drastically over time. This data may not account for this change well.

Below are the resources I will be using to help access this data

In [2]:
import json
import pandas as pd

In [3]:
with open(r"C:\Users\tjl2k\EMAT32110_FA23\liked_posts.json") as f:
    data = json.load(f)

## Loading the Data into Python

The data from Instagram was compiled in a complex manner where it was formatted into multiple lists inside lists with the objects needed hidden deep inside. Further down this notebook you will see how I accessed this data that was hidden deep inside the lists.

In [ ]:
data["likes_media_likes"]

This is placing the data from the original JSON into a Pandas DataFrame to get a survey of the land

In [5]:
introData = pd.DataFrame(data)

In [6]:
introData.head()

likes_media_likes
0  {'title': 'beyondgorilla', 'string_list_data':...
1  {'title': 'officialnbabuzz', 'string_list_data...
2  {'title': 'mirahelsel', 'string_list_data': [{...
3  {'title': 'officialnbabuzz', 'string_list_data...
4  {'title': 'heismans', 'string_list_data': [{'h...

The table below is my inital data set and it is separated into title and another list title "String List Data" with more data I will access further down

In [7]:
likesTable = pd.DataFrame(data["likes_media_likes"])

In [8]:
likesTable.head()

title                                   string_list_data
0    beyondgorilla  [{'href': 'https://www.instagram.com/p/Cwf0sdL...
1  officialnbabuzz  [{'href': 'https://www.instagram.com/p/CwgJKk7...
2       mirahelsel  [{'href': 'https://www.instagram.com/p/CwgL0u1...
3  officialnbabuzz  [{'href': 'https://www.instagram.com/p/Cwfs8wa...
4         heismans  [{'href': 'https://www.instagram.com/reel/CwN0...

Below is a list comprehension used to help me access the timestamp data from within the dictionary titled "string_list_data". I took the data from this and placed it into a seperate column in the table.

In [9]:
timeStampData = [x["string_list_data"][0]["timestamp"] for x in data["likes_media_likes"]]

In [10]:
timeStampTab = pd.DataFrame(timeStampData)

In [11]:
likesTable["Timestamps"] = timeStampTab

In [12]:
likesTable

title                                   string_list_data  \
0       beyondgorilla  [{'href': 'https://www.instagram.com/p/Cwf0sdL...   
1     officialnbabuzz  [{'href': 'https://www.instagram.com/p/CwgJKk7...   
2          mirahelsel  [{'href': 'https://www.instagram.com/p/CwgL0u1...   
3     officialnbabuzz  [{'href': 'https://www.instagram.com/p/Cwfs8wa...   
4            heismans  [{'href': 'https://www.instagram.com/reel/CwN0...   
...               ...                                                ...   
8913    memes_are_mee  [{'href': 'https://www.instagram.com/p/BSjWVsX...   
8914         dbweasel  [{'href': 'https://www.instagram.com/p/BSjdWbL...   
8915      codron_mfc_  [{'href': 'https://www.instagram.com/p/BQBfqnT...   
8916           daquan  [{'href': 'https://www.instagram.com/p/BShKIt9...   
8917     pixieslimmee  [{'href': 'https://www.instagram.com/p/BSeMysZ...   

      Timestamps  
0     1693260183  
1     1693259555  
2     1693259138  
3     1693251925  
4     1693251805  
...          ...  
8913  1491516131  
8914  1491516127  
8915  1491440334  
8916  1491435206  
8917  1491330045  

[8918 rows x 3 columns]

Below I am grouping the information from my table above by account and sorting the values to see which accounts have the most instances or which accounts I have liked the most posts from.

In [13]:
groupedLikes = likesTable.groupby("title").count().sort_values("string_list_data", ascending = False)

In [17]:
groupedLikes.head(50)

string_list_data  Timestamps
title                                             
nocontextnba                       398         398
espn                               367         367
sportscenter                       365         365
wwe                                302         302
bleacherreport                     253         253
nbaonespn                          188         188
detroitpistons                     188         188
nfl                                173         173
leaguealerts                       159         159
lolhylian                          156         156
officialnbabuzz                    152         152
brwrestling                        139         139
noncontextwrestling                112         112
nba                                108         108
houseofhighlights                  107         107
kentstwbb                           97          97
scoobydoofruitsnacks                96          96
inhalemybees                        89          89
scooby                              87          87
kentstmbb                           83          83
daquan                              79          79
everything_pistons                  73          73
barstooldetroit                     73          73
chicagobulls                        69          69
memezar                             66          66
finnbalor                           60          60
kingjames                           59          59
memes_are_mee                       58          58
espnnfl                             52          52
brgridiron                          51          51
rrachelstrawberry                   49          49
christianmccaffrey                  49          49
formshooting                        48          48
blinker.city.history                46          46
ioshmartin                          46          46
brooklynnets                        44          44
kentstathletics                     44          44
nocontextcpba                       44          44
kidquill                            42          42
kentstfootball                      41          41
tntsportswwe                        41          41
rosebednarr                         38          38
wwenxt                              38          38
ckrunner70                          36          36
grandayyyyyy                        35          35
packers                             35          35
worldwar2.memes                     35          35
suns                                34          34
imkingkota                          32          32
destinythegame                      32          32

These lines of code are me messing around with other functions to find the same data on which accounts I've liked the most

In [15]:
accountLikes = pd.DataFrame(likesTable["title"].value_counts())

In [18]:
accountLikes

count
title                  
nocontextnba        398
espn                367
sportscenter        365
wwe                 302
bleacherreport      253
...                 ...
_kingorange           1
disney                1
blumhouse             1
prowrestlingfeed      1
dbweasel              1

[1110 rows x 1 columns]

## Conclusion

In conclusion, my hypothesis of my likings being geared more towards sports-related activities was correct, but not to the extent I had predicted. One major issue with this data is the exact definition of what is considered sports-related content. There are a couple of accounts within the top 50 that do post sports-related memes, but I decided to refrain from including them in the final total. In addition to that, I refrained from including athletes in this compilation, seeing that they are not directly posting information about sports but rather are celebrities. With this in mind, out of my top 50 most liked pages, 26/50 (52%) were "sports"-related  pages. This data does show my changing interests, as I recognized a lot of the old meme accounts on here, and I am sure if the timestamps of each post were looked at, we could see how my interest in that type of content has slowly waned over the years. I believe that in order to take this one step further, taking a peek at accounts I used to follow would help to solidify this data and provide a more concrete analysis.